src/main/java/sysc4806/project24/mini_survey_monkey/WebMvcConfig.java

```diff
@@ -0,0 +1,30 @@
+package sysc4806.project24.mini_survey_monkey;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import sysc4806.project24.mini_survey_monkey.interceptors.SurveyStateInterceptor;
+import sysc4806.project24.mini_survey_monkey.models.State;
+
+import java.util.List;
+
+@Configuration
+public class WebMvcConfig implements WebMvcConfigurer {
+
+    private final SurveyStateInterceptor draftSurveyInterceptor;
+    private final SurveyStateInterceptor nonDraftSurveyInterceptor;
```

Comment:
**Should these be page specific?**

For example, should we have a `createPageInterceptor` and `surveyPageInterceptor`?

My reasoning is these interceptors _feel_ very specific to pages they are being binded two.

_This is probably too nitpicky, sorry, I am just curious why you designed it this way._